File: tests/fixtures/sample.zig
const std = @import("std");
const mem = @import("std").mem;

/// Maximum number of connections allowed.
const max_connections: u32 = 100;

/// Represents a log level.
const LogLevel = enum {
    debug,
    info,
    warning,
    err,
};

/// A 2D point.
const Point = struct {
    x: f64,
    y: f64,

    /// Calculate distance to another point.
    pub fn distance(self: Point, other: Point) f64 {
        const dx = self.x - other.x;
        const dy = self.y - other.y;
        return @sqrt(dx * dx + dy * dy);
    }

    pub fn origin() Point {
        return .{ .x = 0, .y = 0 };
    }
};

/// Manages a network connection.
const Connection = struct {
    host: []const u8,
    port: u16,
    connected: bool,

    /// Creates a new connection.
    pub fn init(host: []const u8, port: u16) Connection {
        return .{
            .host = host,
            .port = port,
            .connected = false,
        };
    }

    /// Establishes the connection.
    pub fn connect(self: *Connection) !void {
        std.debug.print("Connecting to {s}:{d}\n", .{ self.host, self.port });
        self.connected = true;
    }

    pub fn disconnect(self: *Connection) void {
        self.connected = false;
    }

    pub fn isConnected(self: Connection) bool {
        return self.connected;
    }
};

/// Logs a message at the given level.
pub fn log(level: LogLevel, message: []const u8) void {
    _ = level;
    std.debug.print("{s}\n", .{message});
}

/// Processes a list of connections.
pub fn processConnections(connections: []Connection) u32 {
    var count: u32 = 0;
    for (connections) |*conn| {
        conn.connect() catch continue;
        count += 1;
    }
    return count;
}

test "point distance" {
    const p1 = Point{ .x = 0, .y = 0 };
    const p2 = Point{ .x = 3, .y = 4 };
    const d = p1.distance(p2);
    try std.testing.expectEqual(@as(f64, 5.0), d);
}

File: tests/fixtures/sample.vb
Imports System
Imports System.Collections.Generic

''' <summary>
''' Maximum connections allowed.
''' </summary>
Const MaxConnections As Integer = 100

''' <summary>
''' Represents log severity.
''' </summary>
Enum LogLevel
    Debug
    Info
    Warning
    [Error]
End Enum

''' <summary>
''' Interface for serializable objects.
''' </summary>
Interface ISerializable
    Function ToJson() As String
End Interface

''' <summary>
''' Base class with shared functionality.
''' </summary>
Class Base
    Public ReadOnly Property Name As String

    Sub New(name As String)
        Me.Name = name
    End Sub

    Public Function Description() As String
        Return $"{Me.GetType().Name}({Name})"
    End Function

    Private Sub Validate()
        Debug.Assert(Not String.IsNullOrEmpty(Name))
    End Sub
End Class

''' <summary>
''' Manages a network connection.
''' </summary>
Class Connection
    Inherits Base
    Implements ISerializable

    Public Property Port As Integer
    Private _connected As Boolean = False

    Sub New(host As String, Optional port As Integer = 8080)
        MyBase.New(host)
        Me.Port = port
    End Sub

    Public Sub Connect()
        Console.WriteLine($"Connecting to {Name}:{Port}")
        _connected = True
    End Sub

    Public Sub Disconnect()
        _connected = False
    End Sub

    Public Function IsConnected() As Boolean
        Return _connected
    End Function

    Public Function ToJson() As String Implements ISerializable.ToJson
        Return $"{{""host"":""{Name}"",""port"":{Port}}}"
    End Function
End Class

Structure Point
    Public X As Double
    Public Y As Double

    Function Distance(other As Point) As Double
        Dim dx = X - other.X
        Dim dy = Y - other.Y
        Return Math.Sqrt(dx * dx + dy * dy)
    End Function
End Structure

Module Helpers
    Sub LogMessage(level As LogLevel, message As String)
        Console.WriteLine($"[{level}] {message}")
    End Sub
End Module
